mtls/mtls.py

```diff
@@ -886,12 +914,37 @@ def revoke_cert(self, fingerprint, serial_number, common_name):
     def add_user(self, fingerprint, is_admin=False):
         msg = "Signing Add User Request..."
         payload = {
-            "type": "ADMIN" if is_admin else "USER",
             "fingerprint": fingerprint,
-            "signature": str(self.gen_sig(fingerprint.encode("UTF-8"), msg)),
         }
-        response = self.send_request(payload, method="post")
-        response = response.json()
+        if is_admin:
+            payload["admin"] = True
+        signature = self.gen_sig(json.dumps(payload, sort_keys=True), msg)
+        try:
+            server_url = self.config.get(self.server, "url")
+            pgpb64 = base64.b64encode(str(signature).encode("ascii"))
+            response = requests.post(
+                f"{server_url}/users",
+                json=json.loads(json.dumps(payload, sort_keys=True)),
```

Comment:
Why do you load+dumps?

Reply:
This ensures the keys are sorted, which is a consistency mechanism to be able to verify the detached signature to the content provided over the wire

Reply:
Inside of the requests library it's just going to dump again; potentially without `sort_keys` set.
You're wasting (a small) compute encoding it again just to decode it.

```suggestion
                headers={"Content-Type": content_type},
                data=json.dumps(payload, sort_keys=True),
```